### ISSUE TITLE
Cannot upload files from asset-library url

### ISSUE THREAD

**Issue by benjreinhart** [mybigday/react-native-s3#7]
Passing an asset-library:// url to transfer utility is not allowed. Looks like [this issue](https://github.com/aws/aws-sdk-ios/issues/121)?

Anyways, is there an alternative for working with asset-library urls which are returned from the camera roll?

**Comment by brianjd** [comment]
@benjreinhart Did you find an alternative?

**Comment by jhen0409** [comment]
@brianjd, @benjreinhart's [react-native-aws3](https://github.com/benjreinhart/react-native-aws3) can do. :D

I recently had a similar use case for `asset-library://`, I temporary use [react-native-save-asset-library] (https://github.com/wootwoot1234/react-native-save-asset-library), and consider add this behavior to `react-native-s3`.

**Comment by benjreinhart** [comment]
@brianjd yes [react-native-aws3](https://github.com/benjreinhart/react-native-aws3) should work for uploading files with either URL types and has the added benefit of no native dependencies. However, it currently only supports authenticated `PutObject` calls. I've been meaning to add more support because it has the potential to be a very useful library but unfortunately my time has been limited.

**Comment by brianjd** [comment]
@jhen0409 @benjreinhart Thanks for the responses.  But in my case, use of the transferUtility is important, I think it will be a great boost for UX.  So I will experiment with using [react-native-save-asset-library ](https://github.com/wootwoot1234/react-native-save-asset-library), but I am concerned with file management.

**Comment by benjreinhart** [comment]
@brianjd ... what about the transfer utility do you think will be great for UX over another solution? Just curious because if there's something great that I'm missing I'd consider adding it.

**Comment by brianjd** [comment]
@benjreinhart A few reasons, the maturity of aws sdk, the ability to convert to background job and simply monitor from js when in foreground.  I'll be dealing with people uploading 10+ high res images on avg at a time.
